Modules/System/Cryptography Management/src/Xml/SignedXmlImpl.Codeunit.al

```diff
@@ -114,6 +115,27 @@ codeunit 1461 "SignedXml Impl."
     end;
     #endregion
 
+    #region DataObject
+    procedure InitializeDataObject()
+    begin
+        DotNetDataObject := DotNetDataObject.DataObject();
+    end;
+
+    procedure AddObject(DataObjectXmlElement: XmlElement)
+    var
+        XmlDotNetConvert: Codeunit "Xml DotNet Convert";
+        DotNetXmlElement: DotNet XmlElement;
+    begin
+        XmlDotNetConvert.ToDotNet(DataObjectXmlElement, DotNetXmlElement);
+        DotNetDataObject.LoadXml(DotNetXmlElement);
+        AddObject();
+    end;
+
+    local procedure AddObject()
+    begin
+        DotNetSignedXml.AddObject(DotNetDataObject);
+    end;
```

Comment:
I understand this is referring to the KeyInfo

https://github.com/microsoft/ALAppExtensions/blob/7c34747bcf1e3de8abc4e87b0fcd3bb70c0910b0/Modules/System/Cryptography%20Management/src/Xml/SignedXmlImpl.Codeunit.al#L111

But why do we need a local procedure here?

Reply:
good point, I have removed my local procedure and moved the code to global AddObject method as it was unnecessary.